lib/gridstore/src/gridstore.rs

```diff
@@ -218,6 +218,32 @@ impl<V: Blob> Gridstore<V> {
         raw_sections
     }
 
+    /// Read raw value from the pages. Considering that they can span more than one page.
+    fn read_from_pages_sequential(
```

Comment:
This code duplication makes it harder to maintain, I feel like it should be easier to parametrize which memmap to use by introducing an enum and sticking it in the function. Something like:

```rust
enum MmapRead {
    Sequential,
    Random,
}
```

Reply:
I agree. The reason I still decided to go with code duplication was to prevent introducing (more) branching into this very hot method (most likely a premature optimization though)

I'd rather suggest to use a closure to specify which underlying read-method should be used (which is also not 100% clean) but might prevent adding overhead.

Wdyt?

Reply:
A closure sounds fine by me too